andela/ah-jumanji-frontend
# #161966985 Create the notifications system

#### What does this PR do?
Implement functionality for viewing and manipulating notifications

#### Description of Task to be completed?
Add notifications badge, view for notifiv=cations and functionality to mark notifications as read or unread.

#### How should this be manually tested?
```
$ git clone https://github.com/andela/ah-jumanji-frontend.git
$ cd ah-jumanji-frontend
$ git checkout ft-notifications-system-161966985
$ npm install
$ npm start
```
Login and the notifications should appear in the navbar

#### Any background context you want to provide?
notifications are fetched from the API and stored in the IndexDB. This helps retain state across page reloads. The store is then asynchronously updated in the background and the visual components reflect the changes on page reloads.

#### What are the relevant pivotal tracker stories?
[#161966985](https://www.pivotaltracker.com/story/show/161966985)

#### Screenshots (if appropriate)
- badge with no notifications
<img width="1280" alt="image" src="https://user-images.githubusercontent.com/20369585/51754311-040f3200-20cd-11e9-9537-99dc36b2761e.png">

- A view with expanded notifications
<img width="1280" alt="image" src="https://user-images.githubusercontent.com/20369585/51754563-a4655680-20cd-11e9-8fdf-256bc0743d74.png">

- a view with read notifications
<img width="1280" alt="image" src="https://user-images.githubusercontent.com/20369585/51754610-c8289c80-20cd-11e9-9fd5-4e134aa28c0a.png">

- a view with unread notifications
<img width="1280" alt="image" src="https://user-images.githubusercontent.com/20369585/51754475-73852180-20cd-11e9-8511-de601cd6a104.png">


#### Checklist:
